modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/unified_query_builder/ScopeSelector.tsx

```diff
@@ -0,0 +1,225 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {Body, Cell, Head, Row, Table} from '@clayui/core';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import {openSelectionModal} from 'frontend-js-components-web';
+import React, {useContext, useMemo, useState} from 'react';
+
+import ThemeContext from '../../shared/ThemeContext';
+import {STATUS} from '../UQBEditSXPBlueprintForm';
+import CustomPanel from './shared/CustomPanel';
+
+type Status = STATUS.ACTIVE | STATUS.INACTIVE;
+
+type Scope = {
+	externalReferenceCode: string;
+	name: string;
+	status: Status;
+	type: string;
+};
+
+type Sorting = {
+	column: React.Key;
+	direction: 'ascending' | 'descending';
+};
+
+export default function ScopeSelector({
+	scope,
+	setScope,
+}: {
+	scope: Scope[];
+	setScope: (scope: Scope[]) => void;
+}) {
+	const [sort, setSort] = useState<Sorting>();
+
+	const {
+		namespace,
+		selectScopeURL,
+	}: {namespace: string; selectScopeURL: string} = useContext(ThemeContext);
+
+	const filteredScope = useMemo(() => {
+		if (!sort) {
+			return scope;
+		}
+
+		return scope.slice().sort((a, b) => {
+			let cmp = new Intl.Collator('en', {numeric: true}).compare(
+				a[sort.column as keyof Scope],
+				b[sort.column as keyof Scope]
+			);
+
+			if (sort.direction === 'descending') {
+				cmp *= -1;
+			}
+
+			return cmp;
+		});
+	}, [sort, scope]);
```

Comment:
Don't use abbreviations: `cmp`

Can you also document what this does? `Intl.Collator` is not something I typically see so adding documentation would help to quickly understand the purpose of this.

Also it seems concerning that `en` is hardcoded. Should this be localized?